packages/commerce-sdk-react/src/provider.tsx

```diff
@@ -27,55 +28,81 @@ export interface CommerceApiProviderProps extends ApiClientConfigParams {
     proxy: string
     locale: string
     currency: string
+    redirectURI: string
     queryClientConfig?: QueryClientConfig
+    fetchOptions?: ShopperBasketsTypes.FetchOptions
 }
 
 /**
  * @internal
  */
 export const CommerceApiContext = React.createContext({} as ApiClients)
 
+/**
+ * @internal
+ */
+export const AuthContext = React.createContext({} as Auth)
+
 /**
  * Initialize a set of Commerce API clients and make it available to all of descendant components
  *
  * @param props
  * @returns Provider to wrap your app with
  */
 const CommerceApiProvider = (props: CommerceApiProviderProps): ReactElement => {
-    const {children, clientId, organizationId, shortCode, siteId, proxy, queryClientConfig} = props
-
-    // DEBUG: copy access token from browser
-    const headers = {
-        authorization: ''
-    }
+    const {
+        children,
+        clientId,
+        organizationId,
+        shortCode,
+        siteId,
+        proxy,
+        redirectURI,
+        queryClientConfig,
+        fetchOptions
+    } = props
 
     const config = {
         proxy,
-        headers,
         parameters: {
             clientId,
             organizationId,
             shortCode,
             siteId
         },
-        throwOnBadResponse: true
+        throwOnBadResponse: true,
+        fetchOptions
     }
 
-    // TODO: Initialize the api clients immediately, without waiting for an access token.
-    // See template-retail-react-app/app/commerce-api/index.js for inspiration
-    // especially how Proxy class can be used to wait for the access token and inject it to each request header.
-    const apiClients: ApiClients = {
-        shopperBaskets: new ShopperBaskets(config),
-        shopperContexts: new ShopperContexts(config),
-        shopperCustomers: new ShopperCustomers(config),
-        shopperDiscoverySearch: new ShopperDiscoverySearch(config),
-        shopperGiftCertificates: new ShopperGiftCertificates(config),
-        shopperLogin: new ShopperLogin(config),
-        shopperOrders: new ShopperOrders(config),
-        shopperProducts: new ShopperProducts(config),
-        shopperPromotions: new ShopperPromotions(config),
-        shopperSearch: new ShopperSearch(config)
-    }
+    const apiClients = useMemo(() => {
+        return {
+            shopperBaskets: new ShopperBaskets(config),
+            shopperContexts: new ShopperContexts(config),
+            shopperCustomers: new ShopperCustomers(config),
+            shopperDiscoverySearch: new ShopperDiscoverySearch(config),
+            shopperGiftCertificates: new ShopperGiftCertificates(config),
+            shopperLogin: new ShopperLogin(config),
+            shopperOrders: new ShopperOrders(config),
+            shopperProducts: new ShopperProducts(config),
+            shopperPromotions: new ShopperPromotions(config),
+            shopperSearch: new ShopperSearch(config)
+        }
+    }, [clientId, organizationId, shortCode, siteId, proxy])
```

Comment:
Are you missing the `fetchOptions` in the source list?

Reply:
✅ fixed